### PR TITLE
[AO] refactor tutorial for new API

### PR DESCRIPTION
**Pull request by aclegg3**
## Motivation and Context

Refactors URDF_robotics_tutorial.py to use the new object wrapper API + some supporting changes and bug-fixes.

## How Has This Been Tested

Run the script.

## Types of changes

- [x] Docs change / refactoring / dependency upgrade
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

**Review by jturner65 — APPROVED** [review]
lgtm

File: examples/tutorials/URDF_robotics_tutorial.py [hunk 1]
```diff
@@ -525,43 +502,42 @@ def test_ao_recompute_navmesh(make_video=True, show_video=True):
...
-        robot_id = sim.add_articulated_object_from_urdf(robot_file)
+        robot = art_obj_mgr.add_articulated_object_from_urdf(robot_file)
 
         # place the robot root state relative to the agent
-        place_robot_from_agent(sim, robot_id, angle_correction=0)
+        place_robot_from_agent(sim, robot, angle_correction=0)
 
-        root_transform = sim.get_articulated_object_root_state(robot_id)
+        root_transform = robot.transformation
         R = mn.Matrix4.rotation(mn.Rad(3.14), mn.Vector3(0, 1.0, 0))
```
Comment (Skylion007) [inline examples/tutorials/URDF_robotics_tutorial.py:511]: NIT: Surely this should be PI? Just math.PI or something.

File: examples/tutorials/URDF_robotics_tutorial.py [hunk 2]
```diff
@@ -422,10 +403,8 @@ def test_constraints(make_video=True, show_video=True):
                 observations += simulate(sim, dt=3.0, get_frames=make_video)
                 sim.remove_constraint(constraint_id)
                 sim.remove_constraint(constraint_id2)
-                sim.set_object_motion_type(
-                    habitat_sim.physics.MotionType.KINEMATIC, sphere_id
-                )
-                sim.remove_object(active_sphere_id)
+                sphere.motion_type = habitat_sim.physics.MotionType.KINEMATIC
+                sim.remove_object(sphere2.object_id)
```
Comment (Skylion007) [inline examples/tutorials/URDF_robotics_tutorial.py:407]: This call is still deprecated.

File: examples/tutorials/URDF_robotics_tutorial.py [hunk 3]
```diff
@@ -145,9 +139,9 @@ def test_urdf_memory():
     robot_file = urdf_files[robot_key]
     for _sample in range(1000):
 
-        robot_id = sim.add_articulated_object_from_urdf(robot_file)
+        robot = art_obj_mgr.add_articulated_object_from_urdf(robot_file)
         process_memory_tracking.append(get_process_memory_usage())
-        sim.remove_articulated_object(robot_id)
+        sim.remove_articulated_object(robot.object_id)
```
Comment (jturner65) [inline examples/tutorials/URDF_robotics_tutorial.py:144]: This should be accomplished via the articulated object manager.

File: examples/tutorials/URDF_robotics_tutorial.py [hunk 4]
```diff
@@ -373,7 +362,7 @@ def test_constraints(make_video=True, show_video=True):
                 observations += simulate(sim, dt=3.0, get_frames=make_video)
                 sim.remove_constraint(constraint_id)
                 sim.remove_constraint(constraint_id2)
-                sim.remove_articulated_object(robot_id2)
+                sim.remove_articulated_object(robot2.object_id)
```
Comment (jturner65) [inline examples/tutorials/URDF_robotics_tutorial.py:365]: Here too.

File: examples/tutorials/URDF_robotics_tutorial.py (hunk 3 above)
Comment (Skylion007) [reply examples/tutorials/URDF_robotics_tutorial.py:144]: We should considering adding a Python deprecation warning to these calls.

**Pull request merged**